api/src/adapters/aws/sesv2_adapter.py

```diff
@@ -22,22 +25,23 @@ class SESV2Response(BaseModel):
 
 class BaseSESV2Client(ABC, metaclass=ABCMeta):
     @abstractmethod
-    def list_suppressed_destinations(self) -> SESV2Response:
+    def list_suppressed_destinations(self, start_date: datetime | None = None) -> SESV2Response:
         pass
 
 
 class SESV2Client(BaseSESV2Client):
     def __init__(self) -> None:
-        self.client = boto3.client("sesv2")
+        self.client = get_sesv2_client()
 
     def list_suppressed_destinations(self, start_date: datetime | None = None) -> SESV2Response:
         request_params = {}
         if start_date:
             request_params["StartDate"] = start_date
 
         response = self.client.list_suppressed_destinations(**request_params)
```

Comment:
Something to test / follow-up on - I see that there is pagination support in this with a page size + next token. How large of pages can it return / do we need to support calling this paginated? Probably need something?